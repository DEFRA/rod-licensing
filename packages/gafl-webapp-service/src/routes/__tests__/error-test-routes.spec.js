import errorTestRoutes from '../error-test-routes'

describe('Error test route handler', () => {
  describe('error route', () => {
    it('has a return value with a method of GET and path of /buy/throw-error', async () => {
      expect(errorTestRoutes).toMatchSnapshot()
    })
  })

  describe('ERROR_TESTING handler', () => {
    const errorTesting = errorTestRoutes[0].handler

    it.each([
      ['401', 'Unauthorized'],
      ['403', 'Forbidden'],
      ['500', 'Internal Server Error']
    ])('throws the correct error when given %s as the param', async (code, message) => {
      const mockRequest = {
        query: { error: code }
      }
      const mockH = {}

      await expect(errorTesting(mockRequest, mockH)).rejects.toThrow(message)
    })
  })
})
