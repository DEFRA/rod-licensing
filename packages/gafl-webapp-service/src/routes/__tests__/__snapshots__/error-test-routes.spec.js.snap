// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error test route handler error route has a return value with a method of GET and path of /buy/throw-error 1`] = `
Array [
  Object {
    "handler": [Function],
    "method": "GET",
    "path": "/buy/throw-error",
  },
]
`;
